### ISSUE TITLE
Timeline: add author images

### ISSUE THREAD

**Issue by tonyahowe** [LiteratureInContext/LiC-app#304]
Is it possible to automatically pull and add images of authors from the linked data to our timeline? Not sure what the best option is for display, but something consistent and tidy?

**Comment by wsalesky** [comment]
@tonyahowe Where would I be getting these images? Are there links to them in the works?

**Comment by tonyahowe** [comment]
There's nothing directly available in the associated LCNAF or VIAF records, but there is the WikiData page, which seems to be available for most all of the authors. And in the source of those wikidata pages, there's a line (if an image is available) containing a link to the image file: `<meta property="og:image" content="https://upload.wikimedia.org/wikipedia/commons/thumb/c/cc/CassandraAusten-JaneAusten%28c.1810%29_hires.jpg/1200px-CassandraAusten-JaneAusten%28c.1810%29_hires.jpg">` 

Can the first property "og:image" link be used? If no og:image or no wikidata link from the LCNAF, then do nothing? 

![Screenshot 2024-05-15 at 1 06 56 PM](https://github.com/LiteratureInContext/LiC-app/assets/1153464/afdbd5d7-8a72-4ed5-a76c-2ebf821f22fc)
![Screenshot 2024-05-15 at 1 06 50 PM](https://github.com/LiteratureInContext/LiC-app/assets/1153464/1d94859e-feaf-456e-a7e1-6ffdf7d3efd4)

**Comment by wsalesky** [comment]
@tonyahowe This is tricky to do programmatically, the link to the wikimedia record is not in the mads record, it is in the JSON version, but some of them seem to return funky data which trips up my script. I will leave it open, but am going to focus on some other issues for now.

**Comment by tonyahowe** [comment]
Sounds good—thanks!

From: Winona Salesky ***@***.***>
Date: Thursday, May 16, 2024 at 12:47 PM
To: LiteratureInContext/LiC-app ***@***.***>
Cc: Tonya Howe ***@***.***>, Mention ***@***.***>
Subject: Re: [LiteratureInContext/LiC-app] Timeline: add author images (Issue #304)

@tonyahowe<https://github.com/tonyahowe> This is tricky to do programmatically, the link to the wikimedia record is not in the mads record, it is in the JSON version, but some of them seem to return funky data which trips up my script. I will leave it open, but am going to focus on some other issues for now.

**Comment by wsalesky** [comment]
@tonyahowe yes I think that would work

**Comment by tonyahowe** [comment]
I'll update this with a test--haven't gotten back to it yet.

**Comment by tonyahowe** [comment]
Okay, I added a test in line 14 of shelley-frankenstein.xml:              
<persName type="lcnaf" key="n79061063" timeline="notes/Mary-Shelley.jpg">

Is there a way to designate an existing <graphic> tag? For instance, can we add something to a line like this: <graphic url="notes/Mary-Shelley.jpg">? I'm not sure how that would work, but it might be simpler for student authors?

**Comment by wsalesky** [comment]
@tonyahowe I like the second example, but perhaps add a type='timeline' so it is easy to find? 
example: 
`<graphic type="timeline" url="notes/Mary-Shelley.jpg">`

**Comment by tonyahowe** [comment]
Yes, this is a great idea. I'll add that to the xml and send you a line when it's done.

**Comment by tonyahowe** [comment]
Okay-- sorry this has taken so long @wsalesky ! Here is the test case: https://github.com/LiteratureInContext/LiC-data/blob/4f26022e9715ec0185752b9258118a53a0b25d98/data/Shelley/shelley-frankenstein-1818.xml#L678

**Comment by wsalesky** [comment]
Added, let me know what you think.

**Comment by tonyahowe** [comment]
Oh, cool! I think this is looking good, @wsalesky! Thank you!  A couple details: 
- When viewing on mobile, the user has to scroll to see the image--the top part and the timeline part are sort of smooshed. Possible to add space on mobile? If not, no worries.
- Knight Lab has now added a credit at the bottom! So we no longer need the "* Timeline generated with..." statement.

**Comment by tonyahowe** [comment]
Hi, @wsalesky ! The image in the timeline seems to be broken--though the image with the type="timeline" attribute in the xml is working. Can you double check? This is for Frankenstein: https://github.com/LiteratureInContext/LiC-data/blob/5f142f3694685c115d1945fa4ba7b4c91500bd78/data/Shelley/shelley-frankenstein-1818.xml#L678

**Comment by wsalesky** [comment]
@tonyahowe as far as I can tell this is not a legal use of this image, and the website is blocking our request.  But if the image was legal, the code should work. 

https://www.npg.org.uk/business/images#acc71
